hyperspy/tests/io/test_ripple.py

```diff
@@ -0,0 +1,131 @@
+import tempfile
+import os.path
+import shutil
+
+import numpy as np
+import nose.tools as nt
+import numpy.testing as npt
+
+from hyperspy.io import load
+from hyperspy.signals import Signal, EELSSpectrum, EDSTEMSpectrum
+from hyperspy.io_plugins import ripple
+
+
+# Tuple of tuples (data shape, signal_dimensions)
+SHAPES_SDIM = (((3,), (1, )),
+               ((2, 3), (1, 2)),
+               ((2, 3, 4), (1, 2)),
+               )
+
+TMPDIR = tempfile.mkdtemp()
+MYPATH = os.path.dirname(__file__)
+
+nt.assert_equal.__self__.maxDiff = None
+
+
+def test_ripple():
+    try:
+        for dtype in ripple.dtype2keys.keys():
+            for shape, dims in SHAPES_SDIM:
+                for dim in dims:
+                    yield _run_test, dtype, shape, dim
+    finally:
+        shutil.rmtree(TMPDIR)
+
+
+def _get_filename(s):
+    filename = "test_ripple_sdim-%i_ndim-%i_%s.rpl" % (
+        s.axes_manager.signal_dimension,
+        s.axes_manager.navigation_dimension,
+        s.data.dtype.name)
+    return filename
+
+
+def _create_signal(shape, dim, dtype,):
+    data = np.arange(np.product(shape)).reshape(
+        shape).astype(dtype)
+    if dim == 1:
+        if len(shape) > 2:
+            s = EELSSpectrum(data)
+            s.set_microscope_parameters(
+                beam_energy=100.,
+                convergence_angle=1.,
+                collection_angle=10.)
+        else:
+            s = EDSTEMSpectrum(data)
+            s.set_microscope_parameters(
+                beam_energy=100.,
+                live_time=1.,
+                tilt_stage=2.,
+                azimuth_angle=3.,
+                elevation_angle=4.,
+                energy_resolution_MnKa=5.)
+    else:
+        s = Signal(data)
+        s.axes_manager.set_signal_dimension(dim)
+    for i, axis in enumerate(s.axes_manager._axes):
+        axis.offset = i * 0.5
```

Comment:
Though in this case it does not matter, since no real calculations are done. Up to you then, would merge either way once `BaseSignal` is added